apps/src/templates/sectionsRefresh/QuickAssignTable.jsx

```diff
@@ -15,10 +15,42 @@ export default function QuickAssignTable({
   courseOfferings,
   setSelectedCourseOffering,
   updateCourse,
-  sectionCourse
+  sectionCourse,
+  isNewSection
 }) {
   // Key is type of curriculum e.g. 'Course' or 'Module', which is the singular
   // version of the title we want for the column
+
+  useEffect(() => {
+    if (!isNewSection) {
+      // combines all the data into one object
+      const startingData = {
+        ...courseOfferings[marketingAudience][curriculumTypes.course],
+        ...courseOfferings[marketingAudience][curriculumTypes.standalone_unit],
+        ...courseOfferings[marketingAudience][curriculumTypes.module]
+      };
+
+      const headers = Object.keys(startingData);
+      console.log(startingData);
+
+      // iterates over all courses to find the course assigned
+      headers.map(header => {
+        const courseDataByHeaderValues = Object.values(startingData[header]);
+        courseDataByHeaderValues.map(course =>
+          sectionCourse?.courseOfferingId === course.id
+            ? setSelectedCourseOffering(course)
+            : null
+        );
```

Comment:
nit: since this is calling a function, and not returning anything, i think it would be a little clearer to use a regular loop and `if`, like:
```suggestion
        courseDataByHeaderValues.forEach(course => {
          if (sectionCourse?.courseOfferingId === course.id) {
            setSelectedCourseOffering(course);
          }
        });
```
same for the `headers.map` above, too. this might just be my personal preference, but when I see `map`, i tend to assume we're trying to modify and return data formatted in a different way, where `forEach` makes it immediately clear we don't care about a return value and are just doing things inside a loop.

Reply:
Thanks, I really appreciate the explanation behind this proposed change.

Reply:
I think this matches my mental model too^ :)